src/libgit2/index.c

```diff
@@ -3453,6 +3453,13 @@ static int apply_each_file(const git_diff_delta *delta, float progress, void *pa
 		    &match, NULL))
 		return 0;
 
+	/*
+	 * diff returns the directory when all files are ignored; we never
```

Comment:
Is this behavior of diff intentional?
Filtering out trees in function named apply_each_file looks ... weird.